torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -205,7 +207,13 @@ Tensor norm_backward(Tensor grad, const Tensor& self, const optional<Scalar> & p
     self_scaled = self;
     scale_v = grad / norm;
   } else if (std::isinf(p)) {
-    Tensor is_eq_max = (self.abs() == norm).logical_or_(self.isnan().logical_and_(norm.isnan())).type_as(self);
+    Tensor is_eq_max;
+    if (!areAnyTensorSubclassLike({self, norm})) {
+      // optimization: use in-place
+      is_eq_max = (self.abs() == norm).logical_or_(self.isnan().logical_and_(norm.isnan())).type_as(self);
+    } else {
+      is_eq_max = (self.abs() == norm).logical_or_(self.isnan().logical_and(norm.isnan())).type_as(self);
```

Comment:
the or is still inplace though?

Reply:
Oops, thanks for the catch

Reply:
Actually, I'm not sure if this is a problem. It's actually OK that the logical_or_ is in-place, because `self.isnan().logical_and(norm.isnan())` produces a Tensor with both the characteristics of `self` and `norm`

Reply:
You mean `self.abs() == norm` ?

Reply:
`self.abs() == norm` gives a tensor with both the characteristics of self and norm; if one is batched then the final result is batched.

`self.isnan().logical_and(norm.isnan())` does the same thing

So in the end, 
```
(self.abs() == norm) .inplace_( self.isnan().logical_and(norm.isnan()) )
```
is okay

Reply:
I agree that this is fine.

Just want to confirm that the only one that is important here is `self.abs() == nor` (the one being modified inplace) but we don't care aobut `self.isnan().logical_and(norm.isnan())` (even though in this particular case it is always fine)?

Reply:
They're both important I think -- we're doing `a.inplace_(b)` where `a` is `self.abs() == norm` and `b` is `self.isnan().logical_and(norm.isnan())`. In order for this to work, if `b` is e.g. a BatchedTensor, then `a` must be one too